draft-ietf-quic-recovery.md

```diff
@@ -389,6 +398,13 @@ rttvar_sample = abs(smoothed_rtt - adjusted_rtt)
 rttvar = 3/4 * rttvar + 1/4 * rttvar_sample
 ~~~
 
+An endpoint might postpone the processing of acknowledgements when the
+corresponding decryption keys are not immediately available. For example, a
+client might receive an acknowledgement for a 0-RTT packet that it cannot
+decrypt because 1-RTT packet protection keys are not yet available to it. In
+such cases, an endpoint SHOULD ignore such local delays in its round-trip time
+sample until the handshake is confirmed.
```

Comment:
I would prefer this section moved above line 358 (i.e. "When adjusting an RTT sample using peer-reported acknowledgement delays, an endpoint:"), and this SHOULD repeated in one of the bullet points, as we do for the ack-delay due to buffering on the receiver side.

That said, I wouldn't be surprised if you chose to do that as a separate editorial path after merging this PR.

Reply:
I'm fine with moving this further up, before the smoothed_rtt computation, but I don't think we want to include this in that list. That list is about adjusting the RTT sample based on reported ack delay, and it ends with "don't do it if the result is < min_rtt". The local buffering delay is quite different, in that it doesn't need this min_rtt protection. I'm moving it up but leaving it below the adjustment list.